compiler/src/dotty/tools/dotc/reporting/messages.scala

```diff
@@ -2775,7 +2775,11 @@ extends SyntaxMsg(TargetNameOnTopLevelClassID):
 class NotClassType(tp: Type)(using Context)
 extends TypeMsg(NotClassTypeID), ShowMatchTrace(tp):
   def msg(using Context) = i"$tp is not a class type"
-  def explain(using Context) = ""
+  def explain(using Context) =
+    i"""A class type includes classes and traits in a specific order. Defining a class, even an anonymous class,
+        |requires specifying an order for the traits it extends. For example, `A & B` is not a class type because it
+        |doesn't specify which trait takes precedence, A or B. Both `A with B` and `B with A` are class types.
+        |Class types also can't have refinements."""
```

Comment:
For the record, this is currently true, but https://github.com/lampepfl/dotty/pull/19395/commits/36acad9d837691ec350caa081535740a45fea3d1 could change that (but that change would affect the spec of the language in a pretty deep way so we'd have to revisit a lot of documentation anyway).